src/adapters/fs-stream.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from 'fs';
+import * as stream from 'stream';
+
+import FileSystem from './fs';
+
+import { Entry } from '../types/entries';
+import { Pattern } from '../types/patterns';
+
+export default class FileSystemStream extends FileSystem<NodeJS.ReadableStream> {
+	/**
+	 * Use stream API to read entries for Task.
+	 */
+	public read(patterns: string[]): NodeJS.ReadableStream {
+		const filepaths = patterns.map(this.getEntryPath, this);
+
+		const readable = new stream.Readable({ objectMode: true });
+
+		readable._read = () => {
```

Comment:
Looks like `sync` solution. Maybe use `Promise.all` with promises that resolve with `.push` to stream?